src/lib/support/CommonPersistentData.h

```diff
@@ -0,0 +1,83 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *          Contains a class handling creation of linked list of stored persistent data.
+ */
+
+#include <lib/support/PersistentData.h>
+
+namespace chip {
+
+typedef uint8_t entryIndex;
```

Comment:
Having this in a generic chip namespace seems a bit ... generic.

Without reference to any class, why would a `chip entry index` be only limited to a 8-bit value? When seeing the type `entryIndex` I generally would question "index to what kind of entry?".

I would suggesg:
   - naming: types should start with uppercase 
   - placement: put the typedef and constant within its underlying struct (or namespace the whole thing)

Reply:
Regarding the generality of the name, this struct was initially used to create a list of fabrics stored in flash for the group data. I intend to use it for the same purpose for the scene table, however, it could be used to list pretty much anything else so a generic name was used. I do agree that 8 bit is limitative and this is due to the legacy of being used for fabrics as we do not anticipate more than 255 of them on a single device. I am thinking, I could name it "StoredEntryIndex" to clarify, if I want to keep it generic. Otherwise, I do not intend to use this struct differently than it was used in the GroupDataProvider.h, I could refactor it to its initial name: FabricList, and make this type a FabricIndex. Instead I could increase the number of bit but that is dependant on the future use we anticipate for the struct. What do you think about those solutions? 

Regarding the naming and placement, I will correct it asap. I am thinking a namespace might be handy in the event someone wants to add other persistent data objects to the file.